src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -667,41 +687,21 @@ protected StatementSyntax GetStatementContainingInvocationToExtractedMethodWorke
                 }
 
                 var syntaxNode = originalDocument.Root.GetAnnotatedNodesAndTokens(MethodDefinitionAnnotation).FirstOrDefault().AsNode();
-                if (syntaxNode == null || !(syntaxNode is MethodDeclarationSyntax methodDeclaration))
+                var nodeIsMethodOrLocalFunction = syntaxNode is MethodDeclarationSyntax || syntaxNode is LocalFunctionStatementSyntax;
+                if (syntaxNode == null || !nodeIsMethodOrLocalFunction)
```

Comment:
technically the `syntaxNode == null` is unnecessary.  because `null is not a method or local function`.  so the latter check absorbs hte former.